activerecord/lib/active_record/query_logs_formatter.rb

```diff
@@ -3,8 +3,11 @@
 module ActiveRecord
   module QueryLogs
     class LegacyFormatter # :nodoc:
+      attr_reader :sort_pairs
```

Comment:
I don't like this. IMO the formatter should have a single method that takes a Hash and returns a String. 

All the rest should be private API.

Reply:
Yep, agree.